app/controllers/checkout_controller.rb

```diff
@@ -54,6 +54,9 @@ def update
       else
         update_failed
       end
+    elsif @order.errors.messages[:items_removed].present?
```

Comment:
It seems like there's inconsistency between what's expected when an order is being updated from the cart edit page, and what's happening when the order is being updated or validated at the checkout page. Both are running through this same action but they are doing different things. And various other things seem to run through here in both json and html formats.